frontend/src/host_orchestrator/orchestrator/instancemanager.go

```diff
@@ -412,25 +516,31 @@ func (h *fetchCVDHandler) Fetch(buildID, target string) (string, error) {
 		cvdBin:         h.cvdBin,
 		args:           []string{"fetch", buildArg, dirArg},
 	}
-	err := cvdCmd.Run()
-	if err != nil {
-		entry.result = &fetchCVDResult{Error: fmt.Errorf("fetch cvd stage failed: %w", err)}
-		return "", err
-	}
-	entry.result = &fetchCVDResult{OutDir: outDir}
-	return outDir, nil
+	return cvdCmd.Run()
 }
 
-func (h *fetchCVDHandler) getMapEntry(buildID, target string) *fetchCVDMapEntry {
-	h.mapMutex.Lock()
-	defer h.mapMutex.Unlock()
-	key := fmt.Sprintf("%s_%s", buildID, target)
-	entry := h.map_[key]
-	if entry == nil {
-		entry = &fetchCVDMapEntry{}
-		h.map_[key] = entry
+func (h *artifactsManager) downloadWithBuildAPI(outDir, buildID, target string, bundleType artifactsBundleType) error {
+	downloadMap := map[artifactsBundleType][]string{
+		kernelBundleType: {"bzImage"},
 	}
-	return entry
+	var chans []chan error
+	for _, name := range downloadMap[bundleType] {
+		ch := make(chan error)
+		chans = append(chans, ch)
+		go func() {
+			filename := outDir + "/" + name
+			if err := downloadArtifactToFile(h.buildAPI, filename, name, buildID, target); err != nil {
+				ch <- err
+			}
+			close(ch)
+		}()
+	}
+	for _, ch := range chans {
+		for err := range ch {
+			return err
```

Comment:
returning early without reading from all channels has two defects:
1. some errors are lost
2. you'll leave behind a bunch of routines blocked trying to send on the channel

Reply:
true, used multierror here to capture all the errors.